pkg/devices/power_meter.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package devices
+
+import "fmt"
+
+// Joules represents an energy value in SI unit of Joule.
+// The value can additionally have an SI prefix of Micro, Milli etc
+// not captured in this type.
+// ** Do not assume that the value contained is a multiple of 1 Joule **,
+// it could be MicroJoules, or MilliJoules according to the interface
+// or method which returns it.
+type Joules uint64
+
+// String prints Joules withou units
+func (j Joules) String() string {
+	return fmt.Sprintf("%d", j)
+}
+
+// FromMicroJoules converts a value from microjoules (μJ) to joules (J).
+func FromMicroJoules(μj Joules) Joules {
+	return μj / 1_000_000
+}
+
+// ToMicroJoules converts a value from joules (J) to microjoules (μJ).
+func ToMicroJoules(j Joules) Joules {
+	return j * 1_000_000
+}
+
```

Comment:
Could you please add a unit test. 

I think we should create. 
`type MicroJoules uint64` so that we don't inadvertently add Joules + MicroJoules since both are different units.

Reply:
Could we also please add a unit-test?

Reply:
are y ou sure? more types will only create more incompatibility

Reply:
unit test i will add in implementation PR